### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by ddjnw1yu**

### CODE REVIEW

File: app/filter.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    def get_filter_data(self, filter, data):
+        if filter == "DATA TYPES":
+            filter_result = self.generate_mimetypes_filter_data(data)
+        elif filter == "ANATOMICAL STRUCTURE":
+            filter_result = self.generate_anatomy_filter_data(data)
+        elif filter == "SPECIES":
+            filter_result = self.generate_species_filter_data(data)
+        return filter_result
+
+    def generate_mimetypes_filter_data(self, data):
```
Comment (alan-wu) [inline app/filter.py:36]: Part of the generate_mimetypes_filter_data, generate_anatomy_filter_data and generate_species_filter_data are quite similar.  You can probably merge some of them into a method.

**Review by alan-wu — CHANGES_REQUESTED** [review]
I would like to revisit some of the codes in the future but it looks pretty good so far. Thanks.
I think some of the codes in the generate_****_filter methods can be reused.

File: app/filter.py (hunk 1 above)
Comment (ddjnw1yu) [reply app/filter.py:36]: I have made some updates for these three functions. And think of a way to re-generate the gen3 token when 401 happens, but still needs some tests.

**Review by alan-wu — APPROVED** [review]
Looking good.

**Pull request merged**